internal/backends/winit/xdg_color_scheme.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+
+use std::rc::Weak;
+
+use i_slint_core::items::ColorScheme;
+
+use crate::WinitWindowAdapter;
+
+fn xdg_color_scheme_to_slint(value: zbus::zvariant::OwnedValue) -> ColorScheme {
+    match value.downcast_ref::<u32>() {
+        Ok(1) => ColorScheme::Dark,
+        Ok(2) => ColorScheme::Light,
+        _ => ColorScheme::Unknown,
+    }
+}
+
+pub async fn watch(window_weak: Weak<WinitWindowAdapter>) -> zbus::Result<()> {
+    let connection = zbus::Connection::session().await?;
+    let settings_proxy: zbus::Proxy = zbus::proxy::Builder::new(&connection)
+        .interface("org.freedesktop.portal.Settings")?
+        .path("/org/freedesktop/portal/desktop")?
+        .destination("org.freedesktop.portal.Desktop")?
+        .build()
+        .await?;
+
+    let initial_value: zbus::zvariant::OwnedValue =
+        settings_proxy.call("Read", &("org.freedesktop.appearance", "color-scheme")).await?;
```

Comment:
This is wrong btw, the portal initially had a mistake where it would wrap the return value in a yet zvariant::OwnedValue, so it would be `zvariant::OwnedValue(zvariant::OwnedValue(T))` instead of just `zvariant::OwnedValue(T)`. As changing that to fix the portal would break the API, `ReadOne`  was added that behaves as described by the API docs.

See for reference https://github.com/bilelmoussaoui/ashpd/blob/master/src/desktop/settings.rs#L224-L235 

Although, at this point I am not sure why re-implement what ashpd already does and handles for you. Especially that ashpd itself doesn't depend on anything "exotic" and those deps are probably pulled by some other crate you depend on.

Reply:
Interesting. Thanks for spotting! I tested this scenario and found it working, but I completely missed the deprecation notice. I see that it *works* with the nested value because downcast_ref() recurses, but I agree that calling the non-deprecated portal method is cleaner & correct.

You're right about the dependencies, there's nothing fundamentally wrong with ashpd. In fact, if we start using more portal features, I'll advocate in favour of using it again.